src/xacro/__init__.py

```diff
@@ -413,18 +415,27 @@ def process_include(elt, macros, symbols, func):
         except TypeError:
             raise XacroException('namespaces are supported with --inorder option only')
 
-    for filename in get_include_files(filename_spec, symbols):
+    optional = True if optional == "True" else False
+
+    for filename in get_include_files(filename_spec, symbols, optional):
         # extend filestack
         oldstack = push_file(filename)
-        include = parse(None, filename).documentElement
+        try:
```

Comment:
We actually need only one try catch, but we need it here because the `.yaml` "existence" evaluation is done in `get_include_files()` whilst the `.xacro` is done in `parse()`. I have made only one try catch now and remove the optional parameter passing to the `get_include_files`